Spigot-Server-Patches/0551-Mass-Commit.patch

```diff
@@ -0,0 +1,217 @@
+diff --git a/src/main/java/net/minecraft/server/MobSpawnerTrader.java b/src/main/java/net/minecraft/server/MobSpawnerTrader.java
+index 736b794f8268858e2e63a3aafef328b443386989..2466c10a69baab49289587a8894d52e2f316c780 100644
+--- a/src/main/java/net/minecraft/server/MobSpawnerTrader.java
++++ b/src/main/java/net/minecraft/server/MobSpawnerTrader.java
+@@ -41,22 +41,19 @@ public class MobSpawnerTrader implements MobSpawner {
+                 return 0;
+             } else {
+                 this.d = 24000;
+-                if (!worldserver.getGameRules().getBoolean(GameRules.DO_MOB_SPAWNING)) {
++                int i = this.e;
++
++                this.e = MathHelper.clamp(this.e + 25, 25, 75);
++                this.b.h(this.e);
++                if (this.a.nextInt(100) > i) {
+                     return 0;
++                } else if (this.a(worldserver)) {
++                    this.e = 25;
++                    return 1;
+                 } else {
+-                    int i = this.e;
+-
+-                    this.e = MathHelper.clamp(this.e + 25, 25, 75);
+-                    this.b.h(this.e);
+-                    if (this.a.nextInt(100) > i) {
+-                        return 0;
+-                    } else if (this.a(worldserver)) {
+-                        this.e = 25;
+-                        return 1;
+-                    } else {
+-                        return 0;
+-                    }
++                    return 0;
+                 }
++                
+             }
+         }
+     }
+diff --git a/src/main/java/net/minecraft/server/PlayerList.java b/src/main/java/net/minecraft/server/PlayerList.java
+index 9382e8f79e8edec8885c629a36e230fbec50e1fb..fea6a29f7d1aa264ff67f5e5df65bd6811c9c82f 100644
+--- a/src/main/java/net/minecraft/server/PlayerList.java
++++ b/src/main/java/net/minecraft/server/PlayerList.java
+@@ -18,6 +18,8 @@ import java.util.Map;
+ import java.util.Optional;
+ import java.util.Set;
+ import java.util.UUID;
++import java.util.jar.Attributes;
++
+ import javax.annotation.Nullable;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
+@@ -888,9 +890,15 @@ public abstract class PlayerList {
+         if (isRespawn) {
+             cserver.getPluginManager().callEvent(new com.destroystokyo.paper.event.player.PlayerPostRespawnEvent(entityplayer.getBukkitEntity(), location, isBedSpawn));
+         }
++        
++        if(entityplayer1.world.paperConfig.removeAttackCooldown) {
++        	entityplayer1.getAttributeInstance(GenericAttributes.ATTACK_SPEED).addModifier(new AttributeModifier("DisableCooldown", 2700, AttributeModifier.Operation.ADDITION));
++        }
++        
+         // Paper end
+ 
+         // CraftBukkit end
++        
+         return entityplayer1;
+     }
+ 
+@@ -1010,14 +1018,25 @@ public abstract class PlayerList {
+     public IpBanList getIPBans() {
+         return this.l;
+     }
+-
+-    public void addOp(GameProfile gameprofile) {
```

Comment:
Not in scope.

Reply:
> 
> 
> Not in scope.

But that's the commit that adds the Operator API though, which I did mention in my PR, not sure how that isn't in scope